components/style/values/specified/length.rs

```diff
@@ -209,10 +234,17 @@ impl ViewportPercentageLength {
                 (length, cmp::max(viewport_size.width, viewport_size.height)),
         };
 
-        // See bug 989802. We truncate so that adding multiple viewport units
-        // that add up to 100 does not overflow due to rounding differences
-        let trunc_scaled = ((length.0 as f64) * factor as f64 / 100.).trunc();
-        Au::from_f64_au(trunc_scaled)
+        // FIXME: Find a better way to drop the if-condition.
```

Comment:
I added this during rebasing my patch. Manish fixed this for some rounding issues of viewport, but I need more time to check how to combine his patch with mine, so use this if-condition as a work-around for now.